src/control/IntersectionReferenceSystem.ts

```diff
@@ -71,16 +71,29 @@ export class IntersectionReferenceSystem {
     this.endVector = IntersectionReferenceSystem.getDirectionVector(this.interpolators.trajectory, 1 - thresholdDirectionDist, 1);
     this.startVector = this.endVector.map((d: number) => d * -1);
   }
-
   /**
    * Map a length along the curve to intersection coordinates
+   * @param length length along the curve
+   * @param allowExtension allows for projecting beyond the curve
    */
-  project(length: number): number[] {
+  project(length: number, allowExtension: boolean = false): number[] {
     const { curtain } = this.interpolators;
     const l = (length - this._offset) / this.length;
-    // TODO handle points outside
-    if (l < 0 || l > 1) {
-      return [0, 0];
+    if (allowExtension) {
+      if (l > 1) {
+        const excess = l - 1;
+        const refEnd = curtain.getPointAt(1);
+        return [refEnd[0] + excess * this.length, refEnd[1]];
```

Comment:
I'm not sure this is what we're after. If I understand this correctly it will continue in x(displacement) direction for the excess length. IMHO it would be natural to continue in the same direction as the end of well path, especially in the case where MD=0 which would be vertical from the first point in the curtain. Could you please check this with Kjerand?

Reply:
i ran this by him earlier, it is as intended

Reply:
OK. Still think we're mixing up some concepts here and we will need to document and make it very clear why we do it this way as the extension here is along the trajectory and not not along the wellbore path because if some gives in MD = 0, don't set an offset and set allow extension they are most likely going to be surprised by the result.

Reply:
Change name from allowExtension to allowExtrapolation and document that this is not an extension of the curve, but the trajectory